syntax = "proto2";

package spark;

message CreateTunnelRequest {
  optional string username = 1;
};

message CreateTunnelReply {
  // IP to be assigned to the client
  optional string ip = 1;
  // IP of the peer
  optional string peer_ip = 2;
};

message Packet {
  // IP Packet
  optional bytes payload = 1;
};

message Bullet {
  optional CreateTunnelRequest create_tunnel_request = 1;
  optional CreateTunnelReply create_tunnel_reply = 2;
  optional Packet packet = 3;
};

service Spark {
  rpc CreateTunnel(stream Bullet) returns (stream Bullet) {
  }
}
